packages/store/src/adapter/markdown.ts

```diff
@@ -20,111 +21,164 @@ const markdownConvertableFlavours = [
   'affine:database',
 ];
 
+type TraverseContext = {
+  indentDepth: number;
+  insideTheLists: boolean;
+  numberedListCount: number[];
+};
+
 export class MarkdownAdapter extends BaseAdapter<Markdown> {
+  private markdownBuffer = new StringBuilder();
+
   async convertPageSnapshotToAdapterTarget({
     snapshot,
   }: PageSnapshotPayload): Promise<Markdown> {
-    let buffer = '';
-    buffer += `# ${snapshot.meta.title}\n`;
-    buffer += await this.convertBlockSnapshotToAdapterTarget({
-      snapshot: snapshot.block,
-    });
+    const buffer = new StringBuilder();
+    buffer.write(`# ${snapshot.meta.title}\n`);
+    buffer.write(
+      await this.convertBlockSnapshotToAdapterTarget({
+        snapshot: snapshot.block,
+      })
+    );
     return buffer.toString();
   }
   async convertBlockSnapshotToAdapterTarget({
```

Comment:
Can we simplify this API? It's called as `markdownAdapater.convertBlockSnapshotToAdapterTarget`, seems a lot of duplications.

Reply:
Do you have some suggestion?

Reply:
`blockSnapshotToTarget` ?

Reply:
Maybe `fromXXXSnapshot`?